src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,420 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Directive, Input,
+  OnDestroy, AfterViewInit, ElementRef, Optional} from '@angular/core';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+import {extendObject} from '../core/util/object-extend';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+import {createElement} from "@angular/core/src/view/element";
+
+
+/**
+ * Directive that marks an element as a "sticky region", meant to contain exactly one sticky-header
+ * along with the content associated with that header. The sticky-header inside of the region will
+ * "stick" to the top of the scrolling container as long as this region is within the scrolling
+ * viewport.
+ *
+ * If a user does not explicitly define a sticky-region for a sticky-header, the direct
+ * parent node of the sticky-header will be used as the sticky-region.
+ */
+@Directive({
+  selector: '[cdkStickyRegion]',
+})
+export class CdkStickyRegion {
+  constructor(public readonly _elementRef: ElementRef) { }
+}
+
+
+// STICK_START_CLASS is used to mark a header is stuck.
+const STICK_START_CLASS = 'cdk-sticky-header-start';
+// STICK_END_CLASS is used to mark a header is unstuck.
+const STICK_END_CLASS = 'cdk-sticky-header-end';
+/**
+ * Set a debounce time which is used in debounce() function when adding event listeners.
+ * Set is as 5. Because if the DEBOUNCE_TIME is set as a too large number. The sticky effect
+ * during scroll will become vary strange and can not be scrolled smoothly.
+ */
+const DEBOUNCE_TIME: number = 5;
+/**
+ * Directive that marks an element as a sticky-header. Inside of a scrolling container (marked with
+ * cdkScrollable), this header will "stick" to the top of the scrolling viewport while its sticky
+ * region (see cdkStickyRegion) is in view.
+ */
+@Directive({
+  selector: '[cdkStickyHeader]',
+})
+export class CdkStickyHeader implements OnDestroy, AfterViewInit {
+
+  /**
+   * Set the sticky-header's z-index as 10 in default. Make it as an input
+   * variable to make user be able to customize the zIndex when
+   * the sticky-header's zIndex is not the largest in current page.
+   * Because if the sticky-header's zIndex is not the largest in current page,
+   * it may be sheltered by other element when being stuck.
+   */
+  @Input('cdkStickyHeaderZIndex') zIndex: number = 10;
+
+  isStuck: boolean = false;
+  /**
+   * isStickyPositionSupported == true, means current browser support 'position: sticky'
+   * or 'position: -webkit-sticky'.
+   * isStickyPositionSupported == false, means current browser does not support sticky
+   * positioning and need to use the original implementation to get the sticky effect.
```

Comment:
This whole comment can be shortened to 
`Whether the browser support CSS sticky positioning.`